### PR TITLE
raft: do no attch term to MsgReadIndex

### PR DESCRIPTION
**Pull request by mrmiywj**
#1240 
MsgReadIndex, as MsgProp, is to be forwarded to leader.
So we should treat it as local message.
@siddontang  @BusyJay @disksing

### CODE REVIEW

**Comment by siddontang** [comment]
LGTM

File: tests/test_raw_node.rs [hunk 1]
```diff
@@ -100,6 +100,65 @@ fn test_raw_node_step() {
...
+
+    // send readindex request to r2(follower)
+    let _ =
+        nt.peers.get_mut(&2).unwrap().step(new_message_with_entries(2,
+                                                                    2,
+                                                                    MessageType::MsgReadIndex,
+                                                                    vec![test_entries.clone()]));
+
+    // verify r2(follower) forwards this message to r1(leader) with term not set
+    assert_eq!(nt.peers.get(&2).unwrap().msgs.len(), 1);
```
Comment (BusyJay) [inline tests/test_raw_node.rs]: `nt.peers[&2].msgs.len()`

File: tests/test_raw_node.rs [hunk 2]
```diff
@@ -100,6 +100,65 @@ fn test_raw_node_step() {
...
+
+    // send readindex request to r3(follower)
+    let _ =
+        nt.peers.get_mut(&3).unwrap().step(new_message_with_entries(3,
+                                                                    3,
+                                                                    MessageType::MsgReadIndex,
+                                                                    vec![test_entries.clone()]));
+
+    // verify r3(follower) forwards this message to r1(leader) with term not set as well.
+    assert_eq!(nt.peers.get(&3).unwrap().msgs.len(), 1);
```
Comment (BusyJay) [inline tests/test_raw_node.rs]: need to assert the only message is equal to `read_index_msg2`.

**Comment by mrmiywj** [comment]
PTAL @BusyJay

**Comment by BusyJay** [comment]
LGTM

**Pull request merged**